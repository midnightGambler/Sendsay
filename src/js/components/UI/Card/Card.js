import React from "react";

const Card = ({ title, subtitle, className = '', children }) => (
  <div className={`card ${className}`}>
    {title && <h1 className="card__title">{title}</h1>}
    {subtitle && <p className="card__subtitle">{subtitle}</p>}
    {children}
  </div>
);

export default Card;
