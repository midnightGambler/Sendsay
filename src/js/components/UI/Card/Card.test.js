import React from "react";
import Card from "./Card";
import { shallow } from "enzyme";

it("should render card component (no props passed)", () => {
  const card = shallow(
    <Card>
      <div>ha ha business</div>
    </Card>
  );
  expect(card).toMatchSnapshot();
});
it("should render card component with title", () => {
  const card = shallow(
    <Card title="sample title">
      <div>ha ha business</div>
    </Card>
  );
  expect(card).toMatchSnapshot();
});
it("should render card component with subtitle", () => {
  const card = shallow(
    <Card subtitle="sample subtitle">
      <div>ha ha business</div>
    </Card>
  );
  expect(card).toMatchSnapshot();
});
