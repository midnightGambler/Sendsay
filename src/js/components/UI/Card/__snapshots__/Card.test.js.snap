// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card component (no props passed) 1`] = `
<div
  className="card "
>
  <div>
    ha ha business
  </div>
</div>
`;

exports[`should render card component with subtitle 1`] = `
<div
  className="card "
>
  <p
    className="card__subtitle"
  >
    sample subtitle
  </p>
  <div>
    ha ha business
  </div>
</div>
`;

exports[`should render card component with title 1`] = `
<div
  className="card "
>
  <h1
    className="card__title"
  >
    sample title
  </h1>
  <div>
    ha ha business
  </div>
</div>
`;
